<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Quick-start Guide &#8212; DEBrowser v1.30.2 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="../_static/basic.css?v=686e5160" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css?v=27fed22d" />
    <script src="../_static/documentation_options.js?v=00024e28"></script>
    <script src="../_static/doctools.js?v=9bcbadda"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="quick-start-guide">
<h1>Quick-start Guide<a class="headerlink" href="#quick-start-guide" title="Link to this heading">¶</a></h1>
<p>This guide is walkthrough for the DEBrowser from start to finish.</p>
<section id="getting-started">
<h2>Getting Started<a class="headerlink" href="#getting-started" title="Link to this heading">¶</a></h2>
<p>First off, we need to install R package of DEBrowser from bioconductor:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">source</span><span class="p">(</span><span class="s2">&quot;https://bioconductor.org/biocLite.R&quot;</span><span class="p">)</span>
<span class="n">biocLite</span><span class="p">(</span><span class="s2">&quot;debrowser&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>One you have installed the R package, you can call these R commands:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">library</span><span class="p">(</span><span class="n">debrowser</span><span class="p">)</span>
<span class="n">startDEBrowser</span><span class="p">()</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>For more information on installing DEBrowser locally, please consult our <a class="reference external" href="http://debrowser.readthedocs.io/en/latest/local/local.html">Installation Guide</a>.</p>
</div>
<p>Once you’ve made your way to the website, or you have a local instance of DEBrowser running, you will be greeted with data loading section:</p>
<img alt="../_images/data_load.png" class="align-center" src="../_images/data_load.png" />
<p>To begin the analysis, you need to upload your <em>count data file</em> (comma or semicolon separated (CSV), and tab separated (TSV) format) to be analyzed and choose appropriate separator for the file (comma, semicolon or tab).</p>
<p>If you do not have a dataset to upload, you can use the built in demo data file by clicking on the ‘Load Demo (Vernia et al.)!’ button.  To view the entire demo data file, you can download
this <a class="reference external" href="https://bioinfo.umassmed.edu/pub/debrowser/simple_demo.tsv">demo set</a>. For another example, try our <a class="reference external" href="https://bioinfo.umassmed.edu/pub/debrowser/advanced_demo.tsv">full dataset (Vernia et. al)</a> .</p>
<p>The structure of the count data files are shown below:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>gene</p></th>
<th class="head"><p>exp1</p></th>
<th class="head"><p>exp2</p></th>
<th class="head"><p>cont1</p></th>
<th class="head"><p>cont2</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>DQ714</p></td>
<td><p>0.00</p></td>
<td><p>0.00</p></td>
<td><p>0.00</p></td>
<td><p>0.00</p></td>
</tr>
<tr class="row-odd"><td><p>DQ554</p></td>
<td><p>0.00</p></td>
<td><p>0.00</p></td>
<td><p>0.00</p></td>
<td><p>0.00</p></td>
</tr>
<tr class="row-even"><td><p>AK028</p></td>
<td><p>2.00</p></td>
<td><p>1.29</p></td>
<td><p>0.00</p></td>
<td><p>0.00</p></td>
</tr>
</tbody>
</table>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>DEBrowser also accepts count data files via hyperlink, for more information please see the <a class="reference external" href="quickstart.html#autoload-data-via-hyperlink">autoload data via hyperlink</a> section.</p>
</div>
<p>In addition to the count data file; you might need to upload metadata file to correct for batch effects or any other normalizing conditions you might want to address that might be within your results. To handle for these conditions, simply create a metadata file by using the example table at below or download sample file from <a class="reference external" href="https://bioinfo.umassmed.edu/pub/debrowser/simple_demo_meta.txt">this link</a>. Metadata file also simplifies condition selection for complex data. The columns you define in this file can be selected in condition selection page. Make sure you have defined two conditions per column. If there are more than two conditions in a column, those can be defined empty. Please note that, if your data is not complex, metadata file is optional, you don’t need to upload.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>sample</p></th>
<th class="head"><p>batch</p></th>
<th class="head"><p>condition</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>exper_rep1</p></td>
<td><p>1</p></td>
<td><p>A</p></td>
</tr>
<tr class="row-odd"><td><p>exper_rep2</p></td>
<td><p>2</p></td>
<td><p>A</p></td>
</tr>
<tr class="row-even"><td><p>exper_rep3</p></td>
<td><p>1</p></td>
<td><p>A</p></td>
</tr>
<tr class="row-odd"><td><p>control_rep1</p></td>
<td><p>2</p></td>
<td><p>B</p></td>
</tr>
<tr class="row-even"><td><p>control_rep2</p></td>
<td><p>1</p></td>
<td><p>B</p></td>
</tr>
<tr class="row-odd"><td><p>control_rep3</p></td>
<td><p>2</p></td>
<td><p>B</p></td>
</tr>
</tbody>
</table>
<p>Metadata file can be formatted with comma, semicolon or tab separators similar to count data files. These files used to establish different batch effects for multiple conditions.
You can have as many conditions as you may require, as long as all of the samples are present.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The example above would result in the first set of conditions as <code class="docutils literal notranslate"><span class="pre">exper_rep1</span></code>, <code class="docutils literal notranslate"><span class="pre">exper_rep2</span></code>, <code class="docutils literal notranslate"><span class="pre">exper_rep3</span></code> from <code class="docutils literal notranslate"><span class="pre">A</span></code> and second set of conditions as <code class="docutils literal notranslate"><span class="pre">control_rep1</span></code>, <code class="docutils literal notranslate"><span class="pre">control_rep2</span></code>, <code class="docutils literal notranslate"><span class="pre">control_rep3</span></code> from <code class="docutils literal notranslate"><span class="pre">B</span></code> as they correspond to those conditions in the <code class="docutils literal notranslate"><span class="pre">condition</span></code> column.</p>
<p>In the same way, ‘batch’ would have the first set as <code class="docutils literal notranslate"><span class="pre">exper_rep1</span></code>, <code class="docutils literal notranslate"><span class="pre">exper_rep3</span></code>, <code class="docutils literal notranslate"><span class="pre">control_rep2</span></code> from <code class="docutils literal notranslate"><span class="pre">1</span></code> and second set as <code class="docutils literal notranslate"><span class="pre">exper_rep2</span></code>, <code class="docutils literal notranslate"><span class="pre">control_rep1</span></code>, <code class="docutils literal notranslate"><span class="pre">control_rep3</span></code> from <code class="docutils literal notranslate"><span class="pre">2</span></code> as they correspond to those conditions in the <code class="docutils literal notranslate"><span class="pre">batch</span></code> column.</p>
</div>
<p>Once the count data and metadata files have been loaded in DEBrowser, you can click upload button to visualize your data as shown at below:</p>
<img alt="../_images/upload_summary.png" class="align-center" src="../_images/upload_summary.png" />
<p>After loading the gene quantification file, and if specified the metadata file containing your batch correction fields, you then have the option to filter low counts and conduct batch effect correction prior to your analysis. Alternatively, you may skip these steps and directly continue with differential expression analysis or view quality control (QC) information of your dataset.</p>
</section>
<section id="low-count-filtering">
<h2>Low Count Filtering<a class="headerlink" href="#low-count-filtering" title="Link to this heading">¶</a></h2>
<p>In this section, you can simultaneously visualize the changes of your dataset while filtering out the low count genes. Choose your filtration criteria from <strong>Filtering Methods</strong> box which is located just center of the screen. Three methods are available to be used:</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>Max:</strong> Filters out genes where maximum count for each gene across all samples are less than defined threshold.</p></li>
<li><p><strong>Mean:</strong> Filters out genes where mean count for each gene are less than defined threshold.</p></li>
<li><p><strong>CPM:</strong>      First, counts per million (CPM) is calculated as the raw counts divided by the library sizes and multiplied by one million. Then it filters out genes where at least defined number of samples is less than defined CPM threshold.</p></li>
</ul>
</div></blockquote>
<p>After selection of filtering methods and entering threshold value, you can proceed by clicking <strong>Filter</strong> button which is located just bottom part of the <strong>Filtering Methods</strong> box. On the right part of the screen, your filtered dataset will be visualized for comparison as shown at figure below.</p>
<a class="reference internal image-reference" href="../_images/filtering.png"><img alt="../_images/filtering.png" class="align-center" src="../_images/filtering.png" style="width: 99%;" />
</a>
<p>You can easily compare following features, before and after filtering:</p>
<blockquote>
<div><ul class="simple">
<li><p>Number of genes/regions.</p></li>
<li><p>Read counts for each sample.</p></li>
<li><p>Overall histogram of the dataset.</p></li>
<li><p>gene/region vs samples data</p></li>
</ul>
</div></blockquote>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>To investigate the gene/region vs samples data in detail as shown at below, you may click the <strong>Show Data</strong> button, located bottom part of the data tables. Alternatively, you may download all filtered data by clicking <strong>Download</strong> button which located next to <strong>Show Data</strong> button.</p>
</div>
<a class="reference internal image-reference" href="../_images/show_data.png"><img alt="../_images/show_data.png" class="align-center" src="../_images/show_data.png" style="width: 70%;" />
</a>
<p>Afterwards, you may continue your analysis with <strong>Batch Effect Correction</strong> or directly jump to differential expression analysis or view quality control (QC) information of your dataset.</p>
</section>
<section id="batch-effect-correction-and-normalization">
<h2>Batch Effect Correction and Normalization<a class="headerlink" href="#batch-effect-correction-and-normalization" title="Link to this heading">¶</a></h2>
<p>If specified metadata file containing your batch correction fields, then you have the option to conduct batch effect correction prior to your analysis. By adjusting parameters of <strong>Options</strong> box, you can investigate your character of your dataset. These parameters of the options box are explained as following:</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>Normalization Method:</strong> DEBrowser allows performing normalization prior the batch effect correction. You may choose your normalization method (among MRN (Median Ratio Normalization), TMM (Trimmed Mean of M-values), RLE (Relative Log Expression) and upperquartile), or skip this step by choosing <strong>none</strong> for this item. For our sample data, we are going to choose MRN normalization.</p></li>
<li><p><strong>Correction Method:</strong> DEBrowser uses <a class="reference external" href="https://bioconductor.org/packages/release/bioc/vignettes/sva/inst/doc/sva.pdf">ComBat</a> (part of the SVA bioconductor package) or <a class="reference external" href="https://www.bioconductor.org/packages/3.7/bioc/vignettes/Harman/inst/doc/IntroductionToHarman.html">Harman</a> to adjust for possible batch effect or conditional biases. For more information, you can visit following links for documentation: <a class="reference external" href="https://bioconductor.org/packages/release/bioc/vignettes/sva/inst/doc/sva.pdf">ComBat</a>, <a class="reference external" href="https://www.bioconductor.org/packages/3.7/bioc/vignettes/Harman/inst/doc/IntroductionToHarman.html">Harman</a> For our sample data, Combat correction was selected.</p></li>
<li><p><strong>Treatment:</strong> Please select the column that is specified in metadata file for comparison, such as cancer vs control. It is named <em>treatment</em> for our sample metadata.</p></li>
<li><p><strong>Batch:</strong> Please select the column name in metadata file which differentiate the batches. For example in our metadata, it is called <em>batch</em>.</p></li>
</ul>
</div></blockquote>
<p>Upon clicking submit button, comparison tables and plots will be created on the right part of the screen as shown below.</p>
<a class="reference internal image-reference" href="../_images/batch_PCA.png"><img alt="../_images/batch_PCA.png" class="align-center" src="../_images/batch_PCA.png" style="width: 99%;" />
</a>
<a class="reference internal image-reference" href="../_images/batch_IQR.png"><img alt="../_images/batch_IQR.png" class="align-center" src="../_images/batch_IQR.png" style="width: 99%;" />
</a>
<a class="reference internal image-reference" href="../_images/batch_density.png"><img alt="../_images/batch_density.png" class="align-center" src="../_images/batch_density.png" style="width: 99%;" />
</a>
<p>You can investigate the changes on the data by comparing following features:</p>
<blockquote>
<div><ul class="simple">
<li><p>Read counts for each sample.</p></li>
<li><p>PCA, IQR and Density plot of the dataset.</p></li>
<li><p>Gene/region vs samples data</p></li>
</ul>
</div></blockquote>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>You can investigate the gene/region vs samples data in detail by clicking the <strong>Show Data</strong> button, or download all corrected data by clicking <strong>Download</strong> button.</p>
</div>
<p>Since we have completed <strong>batch effect correction and normalization</strong> step, we can continue with one of the following options: ‘Go to DE Analysis’ and, ‘Go to QC plots!’. First option takes you to page where differential expression analyses are conducted with DESeq2, EdgeR or Limma. The second option, ‘Go to QC plots!’, takes you to a page where you can view quality control metrics of your data by PCA, All2All, Heatmap, Density, and IQR plots.</p>
</section>
<section id="de-analysis">
<h2>DE Analysis<a class="headerlink" href="#de-analysis" title="Link to this heading">¶</a></h2>
<p>The first option, ‘Go to DE Analysis’, takes you to the next step where differential expression analyses are conducted.</p>
<blockquote>
<div><ul>
<li><p><strong>Sample Selection:</strong> In order to run DE analysis, you first need to select the samples which will be compared. To do so, click on “Add New Comparison” button, and choose <strong>Select Meta</strong> box as <strong>treatment</strong> to simplify fill <code class="docutils literal notranslate"><span class="pre">Condition</span> <span class="pre">1</span></code> and <code class="docutils literal notranslate"><span class="pre">Condition</span> <span class="pre">2</span></code> based on the <strong>treatment</strong> column of the metadata as shown below.</p>
<blockquote>
<div><img alt="../_images/de_selection.png" class="align-center" src="../_images/de_selection.png" />
</div></blockquote>
</li>
</ul>
<p>If you need to remove samples from a condition, simply select the sample you wish to remove and hit the delete/backspace key. In case, you need to add a sample to a condition you can click on one of the condition text boxes to bring up a list of samples and then click on the sample you wish to add from the list and it will be added to the textbox for that comparison.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>You can add multiple conditions to compare by clicking on “Add New Comparison” button, and view the results separately after DE analysis.</p>
</div>
<ul class="simple">
<li><p><strong>Method Selection:</strong> Three DE methods are available for DEBrowser: DESeq2, EdgeR, and Limma. DESeq2 and EdgeR are designed to normalize count data from high-throughput sequencing assays such as RNA-Seq. On the other hand, Limma is a package to analyse of normalized or transformed data from microarray or RNA-Seq assays. We have selected DESeq2 for our test sample and showed the related results at below.</p></li>
</ul>
</div></blockquote>
<p>After clicking on the ‘Submit!’ button, DESeq2 will analyze your comparisons and store the results into separate data tables. It is important to note that the resulting data produced from DESeq is normalized. Upon finishing the DESeq analysis, a result table will appear which allows you to download the data by clicking “Download” button. To visualize the data with interactive plots please click on “Go to Main Plots!” button.</p>
</section>
<section id="the-main-plots-of-de-analysis">
<h2>The Main Plots of DE Analysis<a class="headerlink" href="#the-main-plots-of-de-analysis" title="Link to this heading">¶</a></h2>
<p>Upon finishing the DESeq analysis, please click on <strong>Go to Main Plots!</strong> button which will open <strong>Main Plots</strong> tab where you will be able to view
the interactive plots.</p>
<img alt="../_images/info_tabs.png" class="align-center" src="../_images/info_tabs.png" />
<p>The page will load with <strong>Scatter Plot</strong>. You can switch to <strong>Volcano Plot</strong> and <strong>MA Plot</strong> by using <strong>Plot Type</strong> section at the left side of the menu. Since these plots are interactive, you can click to <strong>zoom</strong> button on the top of the graph and select the area you would like to zoom in by drawing a rectangle. Please see the plots at below:</p>
<blockquote>
<div><a class="reference internal image-reference" href="../_images/example_main_plots.png"><img alt="../_images/example_main_plots.png" class="align-center" src="../_images/example_main_plots.png" style="width: 99%;" />
</a>
</div></blockquote>
<p><strong>A.</strong> Scatter plot, <strong>B.</strong> Volcano plot, <strong>C.</strong> MA plot</p>
<p>You can easily track the plotting parameters by checking <strong>Plot Information</strong> box as shown at below. Selected DE parameters, chosen dataset, compared conditions, and normalization method are listed. Additionally, heatmap parameters (scaled, centered, log, pseudo-count) could be simply followed by this info box.</p>
<blockquote>
<div><img alt="../_images/plot_information.png" class="align-center" src="../_images/plot_information.png" />
</div></blockquote>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Please keep in mind that to increase the performance of the generating graph, by default 10% of non-significant(NS) genes are used to generate plots. You might show all NS genes by please click <strong>Main Options</strong> button and change Background Data(%) to 100% on the left sidebar.</p>
<a class="reference internal image-reference" href="../_images/example_background_data.png"><img alt="../_images/example_background_data.png" class="align-center" src="../_images/example_background_data.png" style="width: 30%;" />
</a>
</div>
<p>You can hover over the scatterplot points to display more information about the point selected. A few bargraphs will be generated for the user to view as soon as a scatterplot point is hovered over.</p>
<a class="reference internal image-reference" href="../_images/main_plot_hover.png"><img alt="../_images/main_plot_hover.png" class="align-center" src="../_images/main_plot_hover.png" style="width: 99%;" />
</a>
<p><strong>A.</strong> Hover on Fabp3 gene, <strong>B.</strong> Read Counts vs Samples, <strong>C.</strong> Read Counts vs Conditions</p>
<p>You also have a wide array of options when it comes to fold change cut-off levels, p-adjusted (padj) cut-off values, which comparison set to use, and dataset of genes to analyze.</p>
<a class="reference internal image-reference" href="../_images/main_plot_filters.png"><img alt="../_images/main_plot_filters.png" class="align-center" src="../_images/main_plot_filters.png" style="width: 35%;" />
</a>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>It is important to note that when conducting multiple comparisons, the comparisons are labeled based on the order that they are input. If you don’t remember which samples are in your current comparison you can always view the samples in each condition at the top of the main plots.</p>
<img alt="../_images/selected_conditions.png" class="align-center" src="../_images/selected_conditions.png" />
</div>
<p>After DE analysis, you can always download the results in CSV format by clicking the <strong>Download Data</strong> button located under the <strong>Data Options</strong>. You can also download the plot or graphs by clicking on the <strong>download</strong> button at top of each plot or graph.</p>
</section>
<section id="the-heatmap-of-de-analysis">
<h2>The Heatmap of DE Analysis<a class="headerlink" href="#the-heatmap-of-de-analysis" title="Link to this heading">¶</a></h2>
<p>Once you’ve selected a specific region on Main Plots (Scatter, Volcano or MA plot), a new heatmap of the selected area will appear just next to your plot. If you want to hide some groups (such as Up, Down or NS based on DE analysis), just click on the group label on the top right part of the figure. In this way, you can select a specific part of the genes by <strong>lasso select</strong> or <strong>box select</strong> tools that includes only <strong>Up</strong> or <strong>Down</strong> Regulated genes. As soon as you completed your selection, heatmap will be created simultaneously. Please find details about heatmaps on <a class="reference external" href="./../heatmap/heatmap.html">Heatmaps</a> section.</p>
<a class="reference internal image-reference" href="../_images/main_plot_selection.png"><img alt="../_images/main_plot_selection.png" class="align-center" src="../_images/main_plot_selection.png" style="width: 99%;" />
</a>
<p><strong>A.</strong> Box Selection, <strong>B.</strong> Lasso Selection, <strong>C.</strong> Created heatmap based on selection</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>We strongly recommend normalization before plotting heatmaps. To normalize, please change the parameters that are located under: <strong>Data options -&gt; Normalization Methods</strong> and select the method from the dropdown box.</p>
</div>
</section>
<section id="go-term-plots">
<h2>GO Term Plots<a class="headerlink" href="#go-term-plots" title="Link to this heading">¶</a></h2>
<p>The next tab, ‘GO Term’, takes you to the ontology comparison portion of
DEBrowser.  From here you can select the standard dataset options such as
p-adjust value, fold change cut off value, which comparison set to use, and
which dataset to use on the left menu.  In addition to these parameters, you
also can choose from the 4 different ontology plot options: ‘enrichGO’,
‘enrichKEGG’, ‘Disease’, and ‘compareCluster’.  Selecting one of these plot
options queries their specific databases with your current DESeq results.</p>
<img alt="../_images/go_plots_opts.png" class="align-center" src="../_images/go_plots_opts.png" />
<p>Your GO plots include:</p>
<ul class="simple">
<li><p>enrichGO - use enriched GO terms</p></li>
<li><p>enrichKEGG - use enriched KEGG terms</p></li>
<li><p>Disease - enriched for diseases</p></li>
<li><p>compareClusters - comparison of your clustered data</p></li>
</ul>
<p>The types of plots you will be able to generate include:</p>
<p>Summary plot:</p>
<img alt="../_images/go_summary.png" class="align-center" src="../_images/go_summary.png" />
<p>GOdotplot:</p>
<img alt="../_images/go_dot_plot.png" class="align-center" src="../_images/go_dot_plot.png" />
<p>Changing the type of ontology to use will also produce custom parameters for that specific ontology at the bottom of the
left option panel.</p>
<p>Once you have adjusted all of your parameters, you may hit the submit button in the top right and then wait
for the results to show on screen!</p>
</section>
<section id="data-tables">
<h2>Data Tables<a class="headerlink" href="#data-tables" title="Link to this heading">¶</a></h2>
<p>The last tab at the top of the screen displays various different data tables.
These datatables include:</p>
<ul class="simple">
<li><p>All Detected</p></li>
<li><p>Up Regulated</p></li>
<li><p>Down Regulated</p></li>
<li><p>Up+down Regulated</p></li>
<li><p>Selected scatterplot points</p></li>
<li><p>Most varied genes</p></li>
<li><p>Comparison differences</p></li>
</ul>
<img alt="../_images/datatable.png" class="align-center" src="../_images/datatable.png" />
<p>All of the tables tables, except the Comparisons table, contain the following information:</p>
<ul class="simple">
<li><p>ID - The specific gene ID</p></li>
<li><p>Sample Names - The names of the samples given and they’re corresponding tmm normalized counts</p></li>
<li><p>Conditions - The log averaged values</p></li>
<li><p>padj - padjusted value</p></li>
<li><p>log2FoldChange - The Log2 fold change</p></li>
<li><p>foldChange - The fold change</p></li>
<li><p>log10padj - The log 10 padjusted value</p></li>
</ul>
<p>The Comparisons table generates values based on the number of comparisons you have conducted.
For each pairwise comparison, these values will be generated:</p>
<ul class="simple">
<li><p>Values for each sample used</p></li>
<li><p>foldChange of comparison A vs B</p></li>
<li><p>pvalue of comparison A vs B</p></li>
<li><p>padj value of comparison A vs B</p></li>
</ul>
<img alt="../_images/comparisons.png" class="align-center" src="../_images/comparisons.png" />
<p>You can further customize and filter each specific table a multitude of ways.  For unique table or dataset options, select the type of
table dataset you would like to customize on the left panel under ‘Choose a dataset’ to view it’s additional options.
All of the tables have a built in search function at the top right of the table and you can further sort the table
by column by clicking on the column header you wish to sort by.  The ‘Search’ box on the left panel allows for multiple searches via
a comma-separated list.  You can additionally use regex terms such as “^al” or “*lm” for even more advanced searching.
This search will be applied to wherever you are within DEBrowser, including both the plots and the tables.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If you enter more than three lines of genes, search tool will automatically match the beginning and end of the search phrases. Otherwise it will find matched substrings in the gene list.</p>
</div>
<hr class="docutils" />
<p>You can also view specific tables of your input data for each type of dataset available and search for a specific geneset
by inputting a comma-separated list of genes or regex terms to search for in the search box within the left panel.
To view these tables, you must select the tab labeled ‘Tables’ as well as the dataset from the dropdown menu on the left panel.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If you ever want to change your parameters, or even add a new set of comparisons, you can always return to the <em>Data Prep</em> tab to change and resubmit your data.</p>
</div>
</section>
<section id="quality-control-plots">
<h2>Quality Control Plots<a class="headerlink" href="#quality-control-plots" title="Link to this heading">¶</a></h2>
<p>Selecting the ‘QC Plots’ tab will take you to the quality control plots section.  The page opens with a Principal Component Analysis (PCA) plot and users can also view a All2All, heatmap, IQR, and density by choosing <strong>Plot Type</strong> in the left panel. Here the dataset being used in the plots, depends on the parameters you selected in the left panel. Therefore, you are able to adjust the size of the plots under ‘width’ and ‘height’ as well as alter a variety of other parameters to adjust the specific plot you’re viewing.</p>
<p>The All2All plot displays the correlation between each sample, Heatmap shows a heatmap representation of your data, IQR displays a barplot displaying the IQR between samples, and Density will display an overlapping density graph for each sample. You also have the ability to select the type of clustering and distance method for the heatmap produced to further customize your quality control measures. Users also have the option to select which type of normalization methods they would like to use for these specific plotting analysis within the left menu.</p>
<a class="reference internal image-reference" href="../_images/intro_sidebar.png"><img alt="../_images/intro_sidebar.png" class="align-center" src="../_images/intro_sidebar.png" style="width: 30%;" />
</a>
<p>Ploting Options</p>
<img alt="../_images/intro_qc_all2all.png" class="align-center" src="../_images/intro_qc_all2all.png" />
<p>All2All Plot</p>
<img alt="../_images/intro_qc_heatmap.png" class="align-center" src="../_images/intro_qc_heatmap.png" />
<p>Heatmap Options to Normalize All Detected Data and Created Heatmap</p>
<img alt="../_images/intro_qc_pca.png" class="align-center" src="../_images/intro_qc_pca.png" />
<p>PCA Plot</p>
<img alt="../_images/intro_qc_pca_loads.png" class="align-center" src="../_images/intro_qc_pca_loads.png" />
<p>PCA Loadings</p>
<img alt="../_images/iqr_plot.png" class="align-center" src="../_images/iqr_plot.png" />
<p>IQR Plot Before Normalization</p>
<img alt="../_images/iqr_plot_norm.png" class="align-center" src="../_images/iqr_plot_norm.png" />
<p>IQR Plot After Normalization</p>
<img alt="../_images/density_plot.png" class="align-center" src="../_images/density_plot.png" />
<p>Density Plot Before Normalization</p>
<img alt="../_images/density_plot_norm.png" class="align-center" src="../_images/density_plot_norm.png" />
<p>Density Plot After Normalization</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Each QC plot also has options to adjust the plot height and width, as well as a download button for a png output located above each plot.</p>
</div>
<p>For the Heatmap, you can also view an interactive session of the heatmap by selecting the ‘Interactive’ checkbox before submitting your
heatmap request.  Make sure that before selecting the interactive heatmap option that your dataset being used is ‘Up+down’.
Just like in the Main Plots, you can click and drag to create a selection. To select a specific portion of the heatmap, make sure
to highlight the middle of the heatmap gene box in order to fully select a specific gene.  This selection can be used later within the
GO Term plots for specific queries on your selection. For find more details please click <a class="reference external" href="./../heatmap/heatmap.html">Heatmaps</a> section.</p>
<a class="reference internal image-reference" href="../_images/interactive_heatmap.png"><img alt="../_images/interactive_heatmap.png" class="align-center" src="../_images/interactive_heatmap.png" style="width: 99%;" />
</a>
<ol class="upperalpha simple">
<li><p>Before Selection B. Selection of area with zoom tool C. Zoomed heatmap region which allows better viewing resolution.</p></li>
</ol>
</section>
<section id="id4">
<h2>Autoload Data via Hyperlink<a class="headerlink" href="#id4" title="Link to this heading">¶</a></h2>
<p>DEBrowser also accepts TSV’s via hyperlink by following conversion steps. First, using the API provided by Dolphin, we will convert TSV into an html represented TSV using this website:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">dolphin</span><span class="o">.</span><span class="n">umassmed</span><span class="o">.</span><span class="n">edu</span><span class="o">/</span><span class="n">public</span><span class="o">/</span><span class="n">api</span><span class="o">/</span>
</pre></div>
</div>
<p>The two parameters it accepts (and examples) are:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>source=https://bioinfo.umassmed.edu/pub/debrowser/advanced_demo.tsv</p></li>
<li><p>format=JSON</p></li>
</ol>
</div></blockquote>
<p>Leaving you with a hyperlink for:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>https://dolphin.umassmed.edu/public/api/?source=https://bioinfo.umassmed.edu/pub/debrowser/advanced_demo.tsv&amp;format=JSON
</pre></div>
</div>
<p>Next you will need to encode the url so you can pass it to the DEBrowser website.
You can find multiple url encoders online, such as the one located at <a class="reference external" href="https://www.url-encode-decode.com/">this
link.</a>.</p>
<p>Encoding our URL will turn it into this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">http</span><span class="o">%</span><span class="mi">3</span><span class="n">A</span><span class="o">%</span><span class="mi">2</span><span class="n">F</span><span class="o">%</span><span class="mi">2</span><span class="n">Fdolphin</span><span class="o">.</span><span class="n">umassmed</span><span class="o">.</span><span class="n">edu</span><span class="o">%</span><span class="mi">2</span><span class="n">Fpublic</span><span class="o">%</span><span class="mi">2</span><span class="n">Fapi</span><span class="o">%</span><span class="mi">2</span><span class="n">F</span><span class="o">%</span><span class="mi">3</span><span class="n">Fsource</span><span class="o">%</span><span class="mi">3</span><span class="n">Dhttp</span><span class="o">%</span><span class="mi">3</span><span class="n">A</span><span class="o">%</span><span class="mi">2</span><span class="n">F</span><span class="o">%</span><span class="mi">2</span><span class="n">Fbioinfo</span><span class="o">.</span><span class="n">umassmed</span><span class="o">.</span><span class="n">edu</span><span class="o">%</span><span class="mi">2</span><span class="n">Fpub</span><span class="o">%</span><span class="mi">2</span><span class="n">Fdebrowser</span><span class="o">%</span><span class="mi">2</span><span class="n">Fadvanced_demo</span><span class="o">.</span><span class="n">tsv</span><span class="o">%</span><span class="mi">26</span><span class="nb">format</span><span class="o">%</span><span class="mi">3</span><span class="n">DJSON</span>
</pre></div>
</div>
<p>Now this link can be used in DEBrowser as:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">debrowser</span><span class="o">.</span><span class="n">umassmed</span><span class="o">.</span><span class="n">edu</span><span class="p">:</span><span class="mi">443</span><span class="o">/</span><span class="n">debrowser</span><span class="o">/</span><span class="n">R</span><span class="o">/</span>
</pre></div>
</div>
<p>It accepts two parameters:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">1.</span> <span class="n">jsonobject</span><span class="o">=</span> <span class="n">http</span><span class="o">%</span><span class="mi">3</span><span class="n">A</span><span class="o">%</span><span class="mi">2</span><span class="n">F</span><span class="o">%</span><span class="mi">2</span><span class="n">Fdolphin</span><span class="o">.</span><span class="n">umassmed</span><span class="o">.</span><span class="n">edu</span><span class="o">%</span><span class="mi">2</span><span class="n">Fpublic</span><span class="o">%</span><span class="mi">2</span><span class="n">Fapi</span><span class="o">%</span><span class="mi">2</span><span class="n">F</span><span class="o">%</span><span class="mi">3</span><span class="n">Fsource</span><span class="o">%</span><span class="mi">3</span><span class="n">Dhttp</span><span class="o">%</span><span class="mi">3</span><span class="n">A</span><span class="o">%</span><span class="mi">2</span><span class="n">F</span><span class="o">%</span><span class="mi">2</span><span class="n">Fbioinfo</span><span class="o">.</span><span class="n">umassmed</span><span class="o">.</span><span class="n">edu</span><span class="o">%</span><span class="mi">2</span><span class="n">Fpub</span><span class="o">%</span><span class="mi">2</span><span class="n">Fdebrowser</span><span class="o">%</span><span class="mi">2</span><span class="n">Fadvanced_demo</span><span class="o">.</span><span class="n">tsv</span><span class="o">%</span><span class="mi">26</span><span class="nb">format</span><span class="o">%</span><span class="mi">3</span><span class="n">DJSON</span>
<span class="mf">2.</span> <span class="n">title</span><span class="o">=</span> <span class="n">no</span>
</pre></div>
</div>
<p>The finished product of the link will look like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>https://debrowser.umassmed.edu:443/debrowser/R/?jsonobject=https://dolphin.umassmed.edu/public/api/?source=https://bioinfo.umassmed.edu/pub/debrowser/advanced_demo.tsv&amp;format=JSON&amp;title=no
</pre></div>
</div>
<p>Inputting this URL into your browser will automatically load in that tsv to be analyzed by DEBrowser!</p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../index.html">DEBrowser</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2025, Alper Kucukural, Manuel Garber, Onur Yukselen.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.1.3</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="../_sources/quickstart/quickstart.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>