<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Frequently asked questions (FAQ) &#8212; DEBrowser v1.30.2 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="../_static/basic.css?v=686e5160" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css?v=27fed22d" />
    <script src="../_static/documentation_options.js?v=00024e28"></script>
    <script src="../_static/doctools.js?v=9bcbadda"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="frequently-asked-questions-faq">
<h1>Frequently asked questions (FAQ)<a class="headerlink" href="#frequently-asked-questions-faq" title="Link to this heading">¶</a></h1>
<section id="why-un-normalized-counts">
<h2>Why un-normalized counts?<a class="headerlink" href="#why-un-normalized-counts" title="Link to this heading">¶</a></h2>
<p>DESeq2 requires count data as input obtained from RNA-Seq or another high-thorughput sequencing experiment in the form of matrix values. Here we convert un-integer values to integer to be able to run DESeq2. The matrix values should be un-normalized, since DESeq2 model internally corrects for library size. So, transformed or normalized values such as counts scaled by library size should not be used as input. Please use edgeR or limma for normalized counts.</p>
</section>
<section id="why-am-i-getting-error-while-uploading-files">
<h2>Why am I getting error while uploading files?<a class="headerlink" href="#why-am-i-getting-error-while-uploading-files" title="Link to this heading">¶</a></h2>
<ul class="simple">
<li><p>DEBrowser supports tab, comma or semi-colon separated files. However spaces or characters in numeric regions not supported and causes an error while uploading files. It is crutial to remove these kind of instances from the files before uploading files.</p></li>
<li><p>Another reason of getting an error is using same gene name multiple times. This may occurs after opening files in programs such as Excel, which tends to automatically convert some gene names to dates (eg. SEP9 to SEP.09.2018). This leads numerous problems therefore you need to disable these kind of automatic conversion before opening files in these kind of programs.</p></li>
<li><p>Some files contain both tab and space as an delimiter which lead to error. It is required to be cleaned from these kind of files before loading.</p></li>
</ul>
</section>
<section id="why-some-columns-not-showed-up-after-upload">
<h2>Why some columns not showed up after upload?<a class="headerlink" href="#why-some-columns-not-showed-up-after-upload" title="Link to this heading">¶</a></h2>
<p>If a character in numeric area or space is exist in one of your column, either column will be eliminated or you will get an error. Therefore it is crutial to remove for these kind of instances from your files before uploading.</p>
</section>
<section id="why-am-i-getting-error-while-uploading-csv-tsv-files-exported-from-excel">
<h2>Why am I getting error while uploading CSV/TSV files exported from Excel?<a class="headerlink" href="#why-am-i-getting-error-while-uploading-csv-tsv-files-exported-from-excel" title="Link to this heading">¶</a></h2>
<ul class="simple">
<li><p>You might getting an error, because of using same gene name multiple times. This may occurs after opening files in programs such as Excel, which tends to automatically convert some gene names to dates (eg. SEP9 to SEP.09.2018). Therefore you need to disable these kind of automatic conversion before opening files in these kind of programs.</p></li>
</ul>
</section>
<section id="why-can-t-i-see-all-the-background-data-in-main-plots">
<h2>Why can’t I see all the background data in Main Plots?<a class="headerlink" href="#why-can-t-i-see-all-the-background-data-in-main-plots" title="Link to this heading">¶</a></h2>
<p>In order to increase the performance, by default 10% of non-significant(NS) genes are used to generate plots. We strongly suggest you to use all of the NS genes in your plots while publishing your results. You can easily change this parameter by clicking <strong>Main Options</strong> button and change Background Data(%) to 100% on the left sidebar.</p>
</section>
<section id="why-am-i-getting-error-when-i-click-on-de-genes-in-go-term-analysis">
<h2>Why am I getting error when I click on <strong>DE Genes</strong> in Go Term Analysis?<a class="headerlink" href="#why-am-i-getting-error-when-i-click-on-de-genes-in-go-term-analysis" title="Link to this heading">¶</a></h2>
<p>To start <strong>Go Term</strong> analysis, it is important to select correct organism from <strong>Choose an organism</strong> field. After selecting other desired parameters, you can click <strong>Submit</strong> button to run Go Term analysis. After this stage, you will able to see <strong>categories</strong> regarding to your selected gene list in the <strong>Table</strong> Tab. Once you select this category, you can click DE Genes button to see gene list regarding to selected category.</p>
</section>
<section id="how-to-download-selected-data-from-main-plots-qc-plots-heatmaps">
<h2>How to download selected data from Main plots/QC Plots/Heatmaps?<a class="headerlink" href="#how-to-download-selected-data-from-main-plots-qc-plots-heatmaps" title="Link to this heading">¶</a></h2>
<p>First, you need to choose <strong>Choose dataset</strong> field as <strong>selected</strong> under <strong>Data Options</strong> in the left sidebar. When you select this option, new field: <strong>The plot used in selection</strong> will appear under <strong>Choose dataset</strong> field. You need to specify the plot you are interested from following options: Main plot, Main Heatmap, QC Heatmap. Finally you can click <strong>Download Data</strong> button to download data, or if you wish to see the selected data, you can click <strong>Tables</strong> tab.</p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../index.html">DEBrowser</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2025, Alper Kucukural, Manuel Garber, Onur Yukselen.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.1.3</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="../_sources/faq/faq.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>